packages/api/src/command/medical/patient/update-patient.ts

```diff
@@ -1,29 +1,67 @@
 import { Patient, PatientData } from "@metriport/core/domain/patient";
+import { toFHIR } from "@metriport/core/external/fhir/patient/index";
+import { processAsyncError } from "../../../errors";
 import { patientEvents } from "../../../event/medical/patient-event";
+import cqCommands from "../../../external/carequality";
+import cwCommands from "../../../external/commonwell";
+import { upsertPatientToFHIRServer } from "../../../external/fhir/patient/upsert-patient";
 import { PatientModel } from "../../../models/medical/patient";
 import { executeOnDBTx } from "../../../models/transaction-wrapper";
 import { validateVersionForUpdate } from "../../../models/_default";
 import { BaseUpdateCmdWithCustomer } from "../base-update-command";
+import { getFacilityOrFail } from "../facility/get-facility";
+import { addCoordinatesToAddresses } from "./add-coordinates";
 import { getPatientOrFail } from "./get-patient";
 import { sanitize, validate } from "./shared";
-import { addCoordinatesToAddresses } from "./add-coordinates";
 
 type PatientNoExternalData = Omit<PatientData, "externalData">;
 export type PatientUpdateCmd = BaseUpdateCmdWithCustomer &
-  PatientNoExternalData & { externalId?: string };
+  PatientNoExternalData & { externalId?: string; facilityId: string };
 
 // TODO build unit test to validate the patient is being sent correctly to Sequelize
 // See: document-query.test.ts, "send a modified object to Sequelize"
 // See: https://metriport.slack.com/archives/C04DMKE9DME/p1686779391180389
-export const updatePatient = async (
+export async function updatePatient(
   patientUpdate: PatientUpdateCmd,
   emit = true
-): Promise<Patient> => {
+): Promise<Patient> {
+  const { cxId, facilityId } = patientUpdate;
+
+  // validate facility exists and cx has access to it
+  const facility = await getFacilityOrFail({ cxId, id: facilityId });
+
+  const result = await updatePatientWithoutHIEs(patientUpdate, emit);
+
+  const fhirPatient = toFHIR(result);
+  await upsertPatientToFHIRServer(patientUpdate.cxId, fhirPatient);
+
+  // Intentionally asynchronous
+  cwCommands.patient.update(result, facilityId).catch(processAsyncError(`cw.patient.update`));
+
+  // Intentionally asynchronous
+  cqCommands.patient
+    .discover(result, facility.data.npi)
+    .catch(processAsyncError(`cq.patient.update`));
+
+  return result;
+}
+
+export async function updatePatientWithoutHIEs(
+  patientUpdate: PatientUpdateCmd,
+  emit = true
+): Promise<Patient> {
   const { id, cxId, eTag } = patientUpdate;
 
   const sanitized = sanitize(patientUpdate);
   validate(sanitized);
 
+  const addressWithCoordinates = await addCoordinatesToAddresses({
+    addresses: patientUpdate.address,
+    patient: patientUpdate,
+    reportRelevance: true,
+  });
+  if (addressWithCoordinates) patientUpdate.address = addressWithCoordinates;
```

Comment:
There was no reason to have this inside the DB tx, moved to improve performance of the platform as a whole.